nb_train.py

```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+import string
+import re
+import nltk
+import random
+import pickle
+from sklearn.model_selection import train_test_split
+from nltk.corpus import stopwords
+from zipfile import ZipFile
+
+with ZipFile('imdb ratings.zip') as zf:
+    f = zf.open('movie.csv')
+    
+df = pd.read_csv(f)
+smalldf = df[:2000]
+smalldf = smalldf.drop_duplicates(ignore_index = True)
+smalldf.loc[smalldf['label'] == 0, 'label'] = 'neg'
+smalldf.loc[smalldf['label'] == 1, 'label'] = 'pos'
+
+def word_processor(text):  
+    text = "".join([i.lower() for i in text.replace('<br', '') if i not in string.punctuation])
```

Comment:
Yes, I noticed that some review strings have <br></br> tags in them so I removed them. But I do wonder if I need to make my word_processor function capable of processing all kinds of input text that a user might give, which might include other forms of html tags, for eg. How does one go about creating a generic word_processor that can handle any input? 

This in turn is related to a bigger question I have, which is: can my model, which is trained in the context of movie reviews, predict the sentiment of all kinds of strings? Wouldn't context matter in categorizing if a sentence indicate good or bad sentiment?

Reply:
I see.

Regarding your questions:

1) generic word processor

Making a generic word processor is a very large task, since there might be any number of kinds of input. In our case, all we have to do is split the user's input into tokens (aka words). The model might not recognize every word in the input, but that's ok; it'll only use the words it *does* recognize to determine the sentiment.

2) other kinds of strings

Good question. In general, machine learning models have worse performance (aka less accuracy) when executed on data that differs from the data on which they are trained. So to answer your question: probably the model would perform worse on non-movie-reviews than on movie reviews. But there should be enough similarities between the sentiment-laden text in movie reviews and the sentiment-laden text in non-movie-reviews such that it can still do a decent job.